lib/datasource/remote/api/tokenmodels_repository.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:convert';
+
+import 'package:async/async.dart';
+import 'package:collection/collection.dart';
+import 'package:http/http.dart' as http;
+import 'package:seeds/datasource/remote/api/http_repo/http_repository.dart';
+import 'package:seeds/datasource/remote/api/http_repo/seeds_scopes.dart';
+import 'package:seeds/datasource/remote/model/token_model.dart';
+
+/// Retrieve token metadata (used for display of currency cards etc) from
+/// a table in the token master smart contract (tmastr.seeds)
+class TokenModelsRepository extends HttpRepository {
+  Future<Result<List<TokenModel>>> getTokenModels(List<String> acceptList, [List<String>? infoList]) async {
+    /// imports data from all tokens which are accepted under any of the entries in acceptList
+    print("[http] importing token models");
+    final v1ChainUrl = Uri.parse(
+           'https://api.telosfoundation.io/v1/chain/get_table_rows');
+    final idSet = <int>{};
+    final useCaseMap = <int, List<String>>{} ;
+    /// accumulate accepted token id's in idSet
+    /// record valid usecases (from both acceptList and infoList) for each token in useCaseMap
+    for(final useCase in acceptList + (infoList ?? [])) {
+      final String request = '''
+      {
+        "code":"${SeedsCode.accountTokenModels.value}",
+        "table":"acceptances",
+        "scope":"$useCase",
+        "json":true
+      }
+      ''';
+      await http
+          .post(v1ChainUrl, headers: headers, body: request)
+          .then((http.Response response)  {
+              final acceptances = json.decode(response.body)['rows'].toList();
+              final tokenIds = List<int>.from(acceptances.map((row) => row['token_id']).toList());
+              for (final id in tokenIds) {
+                useCaseMap[id] ??= [];
+                useCaseMap[id]!.add(useCase);
+              }
+              if(acceptList.contains(useCase)) {
+                idSet.addAll(tokenIds);
+              }
+          });
+    }
+    final String request = '''
+    {
+      "code":"${SeedsCode.accountTokenModels.value}",
+      "table":"tokens",
+      "scope":"${SeedsCode.accountTokenModels.value}",
+      "json":true
+    }
+    ''';
+    return http
```

Comment:
@gguijarro-c-chwy Is it okay to do 2 services (http calls) in a single method?
Or should we do one service to get the **acceptances** and another for the **tokens**?

Reply:
services functions should only have 1 http call, 
repositories should be calling services. 

But we dont have services, Instead we have our use cases. Our repos are acting as services. 

This getTokenModels should be split into 2 calls. 

then in the useCase we can call both to get the final data you need.